metaflow/plugins/aws/batch/batch_client.py

```diff
@@ -310,6 +311,24 @@ def _register_job_definition(
                 }
             ]
 
+        if efa:
+            if not (isinstance(efa, (int, unicode, basestring))):
+                raise BatchJobException(
+                    "Invalid efa value: ({}) (should be 0 or greater)".format(
+                        efa
+                    )
+                )
+            else:
+                job_definition["containerProperties"]["linuxParameters"]["devices"] = []
+                for i in range(int(efa)):
+                    job_definition["containerProperties"]["linuxParameters"]["devices"] = [
```

Comment:
This job definition seems to follow the single node case in these [docs](https://docs.aws.amazon.com/batch/latest/userguide/efa.html). 

Please correct if I'm misunderstanding, but to function as expected with `@parallel` shouldn't it need a conditional here?
- If single node leave same `job_definition` path and only include the `hostPath` key.
- If multi-node the same vars are set in `job_definition["nodeProperties"]["nodeRangeProperties"]["container"]["linuxParameters"]["devices"]`.

Reply:
@emattia - I may be interpreting this the wrong way, but wouldn't `@parallel` inherit the `job_definition["containerProperties"]` set by the `efa` conditional?

So the user will want to use `@parallel` in conjunction with `efa=1` in the `@batch` decorator

https://github.com/rileyhun/metaflow/blob/cf0cc2f2f66f83a6743a052299a2430b491d695d/metaflow/plugins/aws/batch/batch_client.py#L344C23-L344C23

Reply:
I agree with this end goal:
> So the user will want to use `@parallel` in conjunction with efa=N in the `@batch` decorator

Could you elaborate on what you mean by "`@parallel` inherit the `job_definition["containerProperties"]`"?  

The `@parallel` with `@batch` triggers AWS multinode batch, which ultimately is one job submission through boto3. My comment is because [EFA docs](https://docs.aws.amazon.com/batch/latest/userguide/efa.html) say:

> The location of the [LinuxParameters](https://docs.aws.amazon.com/batch/latest/APIReference/API_LinuxParameters.html) members are different for multi-node parallel jobs and single-node container jobs.

Reply:
I actually missed an `append`, so just made a small fix. Took a look at the job definition, and this looks correct to me
<img width="827" alt="Screenshot 2023-09-15 at 10 39 39 AM" src="https://github.com/Netflix/metaflow/assets/20649466/2707d642-7d57-40be-99fa-ef590c2cc84d">

I think I understand what you're trying to say. I basically need to account for single node cases. 

I was confused and more just pointing out that `@parallel` does most of the correct formatting for the job definition already so the job_definition for efa will be embedded w/in

Reply:
Yep, looks that @parallel is doing the thing I refer to itself, as you point out. Nice!

Reply:
I think it would make sense to disregard the single node use case? If this is meant to be used with `@parallel`, can we assume number of nodes > 1?